tests/test_push.py

```diff
@@ -139,3 +154,249 @@ def test_needs_battery_workaround():
+
+    # Should skip the request
+    assert made_request is False
+    mock_lock.battery.assert_not_called()
+    # State should be unchanged
+    assert result_state == initial_state
+
+
+@pytest.mark.asyncio
+async def test_poll_battery_success():
+    """Test that _poll_battery successfully fetches battery and resets cooldown."""
+    push_lock = PushLock(
+        address="aa:bb:cc:dd:ee:ff",
+        key="0800200c9a66",
+        key_index=1,
+        always_connected=False,
+    )
+    push_lock._name = "Test Lock"
+
+    # Set cooldown to simulate previous timeout
+    push_lock._next_battery_attempt_time = time.monotonic() + 100.0
+
+    mock_lock = MagicMock()
+    battery_state = BatteryState(voltage=6.0, percentage=80)
+    mock_lock.battery = AsyncMock(return_value=battery_state)
+
+    initial_state = LockState(
+        lock=LockStatus.LOCKED,
+        door=DoorStatus.CLOSED,
+        battery=None,
+        auth=None,
+        auto_lock=None,
+        auto_lock_prev=None,
+    )
+
+    # Call _poll_battery (cooldown should be ignored since it's in the future)
+    # Wait a moment to ensure cooldown expires
+    push_lock._next_battery_attempt_time = NEVER_TIME
+
+    result_state, made_request = await push_lock._poll_battery(mock_lock, initial_state)
+
+    # Should make the request
+    assert made_request is True
+    mock_lock.battery.assert_called_once()
+
+    # State should have battery data
+    assert result_state.battery == battery_state
+    assert result_state.auth is not None
+    assert result_state.auth.successful is True
+
+    # Cooldown should be reset to NEVER_TIME
+    assert push_lock._next_battery_attempt_time == NEVER_TIME
+
+
+@pytest.mark.asyncio
+async def test_poll_battery_bleak_error():
+    """Test that _poll_battery handles BleakError gracefully."""
+    push_lock = PushLock(
+        address="aa:bb:cc:dd:ee:ff",
+        key="0800200c9a66",
+        key_index=1,
+        always_connected=False,
+    )
+    push_lock._name = "Test Lock"
+
+    mock_lock = MagicMock()
+    mock_lock.battery = AsyncMock(side_effect=BleakError("Connection failed"))
+
+    initial_state = LockState(
+        lock=LockStatus.LOCKED,
+        door=DoorStatus.CLOSED,
+        battery=None,
+        auth=None,
+        auto_lock=None,
+        auto_lock_prev=None,
+    )
+
+    # Call _poll_battery
+    result_state, made_request = await push_lock._poll_battery(mock_lock, initial_state)
+
+    # Should make the request
+    assert made_request is True
+    mock_lock.battery.assert_called_once()
+
+    # State should be unchanged (error was logged but not raised)
+    assert result_state == initial_state
+
+    # Cooldown should NOT be set (only TimeoutError sets cooldown)
+    assert push_lock._next_battery_attempt_time == NEVER_TIME
+
+
+@pytest.mark.asyncio
+async def test_poll_battery_bleak_dbus_error():
+    """Test that _poll_battery handles BleakDBusError gracefully."""
```

Comment:
nit: ideally we test though the public interfaces, but this lib lacks proper testing infra